### DISCUSSION TITLE
Ocean model grids away from shore and over the land (current intepolation nearshore)

### DISCUSSION THREAD

**Discussion by changks** [OpenDrift/opendrift#1032] [Q&A] [answered]
Hello all,

The operational ocean models usually have a coarse spatial resolution at more than 2 km. That means that model grids could generally be away from shore or over the land. This causes the missing forcings near the coastline, and modeling particles could go on the land, as [this OpenDrift example](https://opendrift.github.io/gallery/example_model_landmask.html) showed.

In reality, we often deal with incidents nearshore. Under the challenges of existing operational ocean models, there are ways to deal with the issue that particles are trapped by the gird boxes of an ocean model, and particles are over the land. 

In the OpenDrift community, what are the practical methods to deal with the current interpolation near the coast?

Please advice! Thank you so much.

Cheers,
Kuo-Hsien

**Comment by gauteh** [comment, marked as answer]
Hi,

if you do not use the landmask from a specific input model a global one is used from the GSHHG shorelines. Data is then interpolated from the coarser ocean model. Usually we add several prioritized readers, so that if a local model doesn't cover the particles a less accurate global one is used.

Regards, Gaute

**Reply by changks** [reply]
May you please point out where the module in OpenDrift handles the interpolation? Thank you, Gaute!

**Reply by gauteh** [reply]
For structured (regularily gridded) input data this class handles spatial interpolation: https://github.com/OpenDrift/opendrift/blob/49fbb8322ba06f49caec29a8ebc7ada59150f627/opendrift/readers/interpolation/structured.py#L85 .
